src/common/http/post_file_request.cpp

```diff
@@ -0,0 +1,42 @@
+#include "post_file_request.hpp"
+
+namespace http {
+PostFile::PostFile(const char *file_path, const char *url_string, const char *file_size)
+    : hdrs {
+        { "Content-Length", file_size, std::nullopt },
+        { nullptr, nullptr, std::nullopt },
+    }
+    , url_string(url_string)
+    , file(fopen(file_path, "rb")) {
+}
+const char *PostFile::url() const {
+    return url_string;
+}
+ContentType PostFile::content_type() const {
+    return ContentType::ApplicationOctetStream;
+}
+Method PostFile::method() const {
+    return Method::Post;
+}
+const HeaderOut *PostFile::extra_headers() const {
+    return hdrs;
+}
+std::variant<size_t, Error> PostFile::write_body_chunk(char *data, size_t size) {
+    if (!file || !size) {
+        return Error::InternalError;
+    }
+
+    size_t written { 0 };
+    while (written < size) {
+        char ch = fgetc(file.get());
+        if (feof(file.get())) {
+            break;
+        }
+        data[written] = ch;
+        ++written;
+    }
```

Comment:
Huh? Why are you doing it by one-character reads? That's probably _very_ ineficient.

Reply:
You're right. I should rethink this.

Reply:
`fread` should do what you want out of the box 😇.